src/ethereum/frontier/vm/instructions/control_flow.py

```diff
@@ -25,3 +36,136 @@ def stop(evm: Evm) -> None:
         The current EVM frame.
     """
     evm.running = False
+
+
+def jump(evm: Evm) -> None:
+    """
+    Alter the program counter to the location specified by the top of the
+    stack.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    InvalidJumpDestError
+        If the jump destination doesn't meet any of the following criteria:
+            * The jump destination is less than the length of the code.
+            * The jump destination should have the `JUMPDEST` opcode (0x5B).
+            * The jump destination shouldn't be part of the data corresponding
+            to `PUSH-N` opcodes.
+    StackUnderflowError
+        If `len(stack)` is less than `1`.
+    OutOfGasError
+        If `evm.gas_left` is less than `8`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_MID)
+    jump_dest = pop(evm.stack)
```

Comment:
Given the `jump_dest` here would always be a `U256` which is a subset of `Uint`. That said, we can check if a U256 is present in a set of Uints for the same reason. So I felt that this conversion is unnecessary. But let me know if converting to Uint actually adds to the readability. Thanks.

Reply:
As long as the set works as expected with `U256` instead of `Uint`, I'm happy to leave as is. I was just concerned since `U256` isn't a subclass of `Uint`.

Reply:
```
In [3]: a1 = {Uint(2), Uint(3), Uint(5), Uint(7), Uint(11), Uint(13)}

In [4]: Uint(5) in a1
Out[4]: True

In [5]: U256(5) in a1
Out[5]: True

In [6]: Uint(9) in a1
Out[6]: False

In [7]: U256(9) in a1
Out[7]: False

```